website/content/docs/operations/nomad-agent.mdx

```diff
@@ -82,22 +82,44 @@ There are several important messages that `nomad agent` outputs:
 
 ## Stopping an Agent
 
-An agent can be stopped in two ways: gracefully or forcefully. By default,
-any signal to an agent (interrupt, terminate, kill) will cause the agent
-to forcefully stop. Graceful termination can be configured by either
-setting `leave_on_interrupt` or `leave_on_terminate` to respond to the
+An agent can be stopped in two ways: gracefully or forcefully. By default, any
+stop signal to an agent (interrupt, terminate, kill) will cause the agent to
+forcefully stop. Graceful termination can be configured by either setting
+[`leave_on_interrupt`][] or [`leave_on_terminate`][] to respond to the
 respective signals.
 
-When gracefully exiting, clients will update their status to terminal on
-the servers so that tasks can be migrated to healthy agents. Servers
-will notify their intention to leave the cluster which allows them to
-leave the [consensus](/nomad/docs/concepts/consensus) peer set.
+When gracefully exiting, servers will notify their intention to leave the
+cluster which allows them to leave the [consensus][] peer set.
 
-It is especially important that a server node be allowed to leave gracefully
-so that there will be a minimal impact on availability as the server leaves
-the consensus peer set. If a server does not gracefully leave, and will not
-return into service, the [`server force-leave` command](/nomad/docs/commands/server/force-leave)
-should be used to eject it from the consensus peer set.
+It is especially important that a server node be allowed to leave gracefully so
+that there will be a minimal impact on availability as the server leaves the
+consensus peer set. If a server does not gracefully leave, and will not return
+into service, the [`server force-leave` command][] should be used to eject it
+from the consensus peer set.
+
+## Signal Handling
+
+In addition to the optional handling of interrupt (`SIGINT`) and terminate
+signals (`SIGTERM`) described in [Stopping an Agent][#stopping-an-agent], Nomad
+supports special behavior for several other signals useful for debugging.
+
+* `SIGUSR1` will cause Nomad to print its [metrics][] without stopping the
+  agent.
+* `SIGQUIT`, `SIGILL`, `SIGTRAP`, `SIGABRT`, `SIGSTKFLT`, `SIGEMT`, or `SIGSYS`
```

Comment:
I see SIGTERM as mentioned above, and SIGHUP which isn't mentioned, here:

https://github.com/hashicorp/nomad/blob/v1.5.3/command/agent/command.go#L890

for the `agent` command signal handling, but I'm not finding these other signals handled anywhere explicitly..  Am I missing something obvious, or something implicit that I'm not aware of?

Reply:
The `SIGUSR1` behavior is provided by the go-metrics library (see [`inmem_signal.go`](https://github.com/hashicorp/go-metrics/blob/master/inmem_signal.go), which I kinda don't _love_ but :shrug: ), and the signals listed in the 2nd bullet point are covered by the Go runtime. See [Default behavior of signals in Go programs](https://pkg.go.dev/os/signal#hdr-Default_behavior_of_signals_in_Go_programs). I'm not documenting the system default handlers for things like job control because that doesn't seem all that useful?

Reply:
Add a bullet point for `SIGHUP` with a link to where we define that behavior in more detail.